.circleci/config.yml

```diff
@@ -95,33 +202,46 @@ commands:
           command: |
             npm config set scripts-prepend-node-path true
             npm install
-            npm install -g sfdx-cli@7.45.1
+            npm install -g sfdx-cli@<< pipeline.parameters.cli-version >>
             npm install -g codecov
             npm install -g junit-merge
       - run:
           name: 'Compile'
           shell: bash.exe
           command: npm run compile
-      - save_cache:
-          key: dependency-cache-win3-{{ checksum "package.json" }}
-          paths:
-            - node_modules
+      - save_dependency_cache:
+          base-cache-path: 'dependency-cache-win3-'
       - run:
           name: 'Run lint'
           shell: bash.exe
           command: |
             npm run lint
 
+executors:
+  default-executor:
+    parameters:
+      # We are unable to use the recommended cimg docker images since they use the root path instead of home
```

Comment:
### Recommended Change from CircleCi - update to cimg docker image

### Conclusion: Not incorporating recommendation
I do think that there is a way that we could still upgrade this image but I'm not convinced it's worth pursuing. The motivation towards changing the docker image is that it should be faster according to [here](https://circleci.com/blog/announcing-our-next-generation-convenience-images-smaller-faster-more-deterministic/). However from doing testing the jobs that it was running all took longer.

### Additional Info from Testing
For using the cimg docker images - [here](https://app.circleci.com/pipelines/github/forcedotcom/salesforcedx-vscode/1440/workflows/d27b1179-5686-4639-8cec-33cebd0e3b1d/jobs/10373) is an example run showing the path that the image uses. This causes compilation problems because of the path. Changing the working directory doesn't make any difference.

Reply:
While not necessarily faster, I think they might be better maintained moving forward. So might be something to consider moving forward.

Reply:
@vazexqi Agreed. From their documentation it looks like node images are in the process of being deprecated. We're going to have to play around with the image more before we can use it do to those path issues. There was probably something small that I was missing